internal/resources/metal/virtual_circuit/resource.go

```diff
@@ -0,0 +1,459 @@
+				Optional:    true,
+				Computed:    true,
+				// TODO: implement SuppressDiffFunc for input with units to bps without units
+			},
+			"tags": {
+				Type:        schema.TypeList,
+				Description: "Tags attached to the virtual circuit",
+				Optional:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"nni_vlan": {
+				Type:        schema.TypeInt,
+				Description: "Equinix Metal network-to-network VLAN ID (optional when the connection has mode=tunnel)",
+				Optional:    true,
+				ForceNew:    true,
+			},
+			"vlan_id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "UUID of the VLAN to associate",
+				ExactlyOneOf: []string{"vlan_id", "vrf_id"},
+			},
+			"vrf_id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "UUID of the VRF to associate",
+				ExactlyOneOf: []string{"vlan_id", "vrf_id"},
+				ForceNew:     true,
+			},
+			"peer_asn": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				RequiredWith: []string{"vrf_id"},
+				Description:  "The BGP ASN of the peer. The same ASN may be the used across several VCs, but it cannot be the same as the local_asn of the VRF.",
+				ForceNew:     true,
+			},
+			"subnet": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				RequiredWith: []string{"vrf_id"},
+				Description: `A subnet from one of the IP blocks associated with the VRF that we will help create an IP reservation for. Can only be either a /30 or /31.
+				 * For a /31 block, it will only have two IP addresses, which will be used for the metal_ip and customer_ip.
+				 * For a /30 block, it will have four IP addresses, but the first and last IP addresses are not usable. We will default to the first usable IP address for the metal_ip.`,
+			},
+			"metal_ip": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				RequiredWith: []string{"vrf_id"},
+				Description:  "The Metal IP address for the SVI (Switch Virtual Interface) of the VirtualCircuit. Will default to the first usable IP in the subnet.",
+			},
+			"customer_ip": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				RequiredWith: []string{"vrf_id"},
+				Description:  "The Customer IP address which the CSR switch will peer with. Will default to the other usable IP in the subnet.",
+			},
+			"md5": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Sensitive:   true,
+				Description: "The password that can be set for the VRF BGP peer",
+			},
+
+			"vnid": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "VNID VLAN parameter, see https://metal.equinix.com/developers/docs/networking/fabric/",
+			},
+			"nni_vnid": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "Nni VLAN ID parameter, see https://metal.equinix.com/developers/docs/networking/fabric/",
```

Comment:
There are a few metal.equinix.com URLs referenced here. I could ignore these as migrated and a separate problem, but this is effectively 90% new code.

This link should be: https://deploy.equinix.com/developers/docs/metal/interconnections/introduction/

The same link is above and may appear elsewhere.

Reply:
90% is a high estimate; this is [mostly the same code](https://github.com/equinix/terraform-provider-equinix/pull/700/commits/0447951f245151a5b35c5d23f9a225ffcd4421d3#diff-a559b25bbfeecdf929180963d85161e3d214577dcc1f671f09d74e2a31f4c838) but moving it made GitHub lose the plot.

That said, I'm fully on board with fixing a couple outdated URLs while I'm in here.